### ISSUE TITLE
Create a completely new subs feed

### ISSUE THREAD

**Issue by OsaSoft** [OsaSoft/youtube-better-subscriptions#30]
Heavily inspired by https://github.com/CodeNinjaUK/youtube-subs-grid
In order to be able to manipulate the feed much more easily, creating a completely new one is the best solution.

Long term stretch goal for a version 2.0

**Comment by DavidDrezen** [comment]
This would be awesome, and it would fix a problem with the current subs feed : if you go too far in the past, YT doesn't show all videos, they show only the most viewed videos (min 500k views).

**Comment by DavidDrezen** [comment]
Yeah I was thinking about crawling each channels (I never had the time to do it by myself). To avoid performance problems, I thought using the local storage to keep the data so we don't have to crawl again the whole channel (maybe it's a bad idea).

**Comment by JeffJacobson** [comment]
I don't know if this is useful information, but YouTube channels have their own RSS feeds.

The TypeScript code below is from a [repo of mine][bookmarklets] that converts a YouTube feed URL into its RSS Feed equivalent. This repo is under The Unlicense so you can use it without restriction as you see fit.

```typescript
/**
 * Converts a YouTube channel URL into an RSS feed URL.
 * @param channelUrl A YouTube channel URL.
 * @returns A {@link URL} for the current channel.
 */
export function convertToFeedLink(channelUrl: string | URL) {
  // https://www.youtube.com/channel/UC25J6ueIa1L2NTqbbAeGN7A
  // https://www.youtube.com/feeds/videos.xml?channel_id=UC25J6ueIa1L2NTqbbAeGN7A

  const url = typeof channelUrl === "string" ? new URL(channelUrl) : channelUrl;

  // Get the end part of the URL which will be the channel ID.
  const parts = url.pathname.split("/");
  const channelId = parts[parts.length - 1];

  // Create output URL.
  return new URL(`feeds/videos.xml?channel_id=${channelId}`, url);
}
```

[bookmarklets]:https://github.com/JeffJacobson/bookmarklets/

**Comment by grandslammer** [comment]
I think this would be a fantastic feature. Not sure if it should be incorporated into the current extension, or create a completely new repo. Maybe create a separate website (e.g. myyoutubesubscriptions.com) where a user would signin with their Google/YT account, and the website would automatically gather their subscriptions and present them to the user in an optimized layout.

This would also potentially act as a fix for the seemingly never-ending UI issues such as the random size thumbnails, or other issues which keep happening whenever there is a new YT update. We would potentially have full control over the UI without having to care about just making it work with YouTube's UI.
